Add event timezone in Model\Event::getHTML

MrPetovan opened the pull request:
Removes the ambiguity described in #10914 

Also fixes a long-standing issue accessing backend modules with an existing session with 2FA activated.

MrPetovan committed:
Add timezone to event dates format in Model\Event::getHTML

- Adds timezone information for user-less contexts like syndication feeds

nupplaphil commented on view/lang/C/messages.po:
```diff
@@ -4167,6 +4166,10 @@ msgstr ""
 msgstr ""
 
+#: src/Model/Event.php:853 src/Module/Debug/Localtime.php:38
+msgid "l F d, Y \\@ g:i A"
```
Is it intended that you don't change this Datetime patterns as well?

MrPetovan replied:
It isn't intended so much as "overlooked". Since it isn't used in event display in feeds, I didn't know it was there.

MrPetovan replied:
After looking into both places, they are meant to show local time and as such do not need to be updated.

annando commented on src/Model/Event.php:
```diff
@@ -49,7 +49,7 @@ public static function getHTML(array $event, $simple = false, $uriid = 0)
 		$uriid = $event['uri-id'] ?? $uriid;
 
-		$bd_format = DI::l10n()->t('l F d, Y \@ g:i A'); // Friday January 18, 2011 @ 8 AM.
+		$bd_format = DI::l10n()->t('l F d, Y \@ g:i A \G\M\T'); // Friday January 18, 2011 @ 8 AM.
```
Can you update the example as well?

MrPetovan replied:
Done.

tobiasd replied:
And does this translate the timezone into the correct local time or will the users have to ask their fav. search engine to translate the times?

MrPetovan committed:
Skip session authentication for backend modules

- This was causing errors accessing these modules with an existing session with 2fa enabled while anonymous calls were going through

MrPetovan committed:
Update main translation file after string update

MrPetovan commented:
After reviewing the usage more carefully, I found out that this format is used in a localized time string, so it is using a potentially variable timezone. I added this information since it can be the timezone of the server itself.
